### PR TITLE
Fix metric-postgres-graphite.rb replication function names for PostgreSQL 10.x+

### PR DESCRIPTION
**Pull request by jfineberg**
## Pull Request Checklist

This will address Issue #52 and provide graphite metrics support in PostgreSQL 10.x+

Refactor metric-postgres-graphite.rb to operate like check-postgres-replication.rb using common methods provided in lib.

-- performs a version check and uses the appropriate function names for getting replication status.

### CODE REVIEW

File: bin/metric-postgres-graphite.rb [hunk 1]
```diff
@@ -64,6 +65,12 @@ class CheckpostgresReplicationStatus < Sensu::Plugin::Metric::CLI::Graphite
          long: '--password=VALUE',
          description: 'Database password'
 
+  option :ssl,
```
Comment (majormoses) [inline bin/metric-postgres-graphite.rb:68]: adding this new option should be called out in the changelog  in a `### Added` header.

File: bin/metric-postgres-graphite.rb [hunk 2]
```diff
@@ -64,6 +65,12 @@ class CheckpostgresReplicationStatus < Sensu::Plugin::Metric::CLI::Graphite
...
+         short: '-S',
```
Comment (majormoses) [inline bin/metric-postgres-graphite.rb:69]: I think this should be `-S VALUE`

File: bin/metric-postgres-graphite.rb [hunk 3]
```diff
@@ -64,6 +65,12 @@ class CheckpostgresReplicationStatus < Sensu::Plugin::Metric::CLI::Graphite
...
+         long: '--ssl',
```
Comment (majormoses) [inline bin/metric-postgres-graphite.rb:70]: I think this should be `--ssl VALUE`

**Review by majormoses — CHANGES_REQUESTED** [review]
Hey thanks for picking this back up, I think it looks good can you show the input and output of the check demonstrating the fix? I don't run postgres so I don't have somewhere to easily test some of the features. Couple of comments but otherwise looks good.

**Comment by phumpal** [comment]
>  I think it looks good can you show the input and output of the check demonstrating the fix

@jfineberg do you need help with this. Would love to see this merged!

**Comment by phumpal** [comment]
@jfineberg gentle ping.

**Comment by phumpal** [comment]
@majormoses FYI

> I think it looks good can you show the input and output of the check demonstrating the fix

Using PostgreSQL 11

```
sensu@postgres-main-slave:~$ metric-postgres-graphite.rb -d airbrake_development -m 10.2.13.11 -s 10.2.3.9 -u<REDACTED> -p<REDACTED> -P 6543
postgres-main-slave.postgres.replication_lag 0 1552085400
```

EDIT: removed the username and password. @majormoses

**Comment by majormoses** [comment]
This looks good, I will work on merging and cutting a release shortly. @phumpal thank you for once again being my postgres savior for getting stuff over the line! just a friendly reminder that putting your DB credentials on the CLI like that exposes the credentials to the process table making it accessible to any user on the system. It's is recommended that you either supply an env var to the sensu process, use token substitution, or do something like this:
```
metric-postgres-graphite.rb -d airbrake_development -m 10.2.13.11 -s 10.2.3.9 -u$(cat my_db_username_file) -p$(my_database_password_file) -P 6543
```

**Comment by phumpal** [comment]
> This looks good, I will work on merging and cutting a release shortly

:+1:

> thank you for once again being my postgres savior for getting stuff over the line!

Of course. Happy to help!

> It's is recommended that you either supply an env var 

My dev environment thanks you ;)

**Review by majormoses — APPROVED** [review]

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-postgres/versions/2.3.2

**Comment by phumpal** [comment]
🍻 thanks!

**Pull request merged**